tests/integration/devfile/cmd_devfile_watch_test.go

```diff
@@ -31,105 +31,61 @@ var _ = Describe("odo devfile watch command tests", func() {
 		helper.CommonAfterEach(commonVar)
 	})
 
-	Context("when running help for watch command", func() {
+	When("running help for watch command", func() {
 		It("should display the help", func() {
 			appHelp := helper.Cmd("odo", "watch", "-h").ShouldPass().Out()
 			helper.MatchAllInOutput(appHelp, []string{"Watch for changes", "git components"})
 		})
 	})
 
-	Context("when executing watch without pushing a devfile component", func() {
-		It("should fail", func() {
+	When("executing watch without pushing a devfile component", func() {
```

Comment:
The `When` contents should explain what is doing the `BeforeEach` part only, here something similar to `When("a component is created"`